ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go

```diff
@@ -119,39 +119,33 @@ func update(c *cli.Context, networkMode, serviceName, clusterName string, cfnCli
 
 func delete(c *cli.Context, cfnClient cloudformation.CloudformationClient, serviceName, projectName, clusterName string) error {
 	sdsStackName := cfnStackName(serviceDiscoveryServiceStackNameFormat, clusterName, serviceName)
-	err := deleteStackIfExisting(sdsStackName, projectName, "Service Discovery Service", cfnClient)
+	err := deleteStack(sdsStackName, projectName, "Service Discovery Service", cfnClient, true, false)
 	if err != nil {
 		return err
 	}
 
 	if c.Bool(flags.DeletePrivateNamespaceFlag) {
 		namspaceStackName := cfnStackName(privateDNSNamespaceStackNameFormat, clusterName, serviceName)
-		err = deleteExistingStack(namspaceStackName, projectName, "Private DNS Namespace", cfnClient)
+		err = deleteStack(namspaceStackName, projectName, "Private DNS Namespace", cfnClient, false, false)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func deleteExistingStack(stackName, projectName, resource string, cfnClient cloudformation.CloudformationClient) error {
+func deleteStack(stackName, projectName, resource string, cfnClient cloudformation.CloudformationClient, ignoreValidation, cleanUp bool) error {
```

Comment:
I don't love requiring so many parameters to a single function -- seems like a good opportunity to consider where the separation of concerns is, e.g. if some of the business logic can be handled by caller.